python-lib/spell_checker/spell_checker.py

```diff
@@ -0,0 +1,127 @@
+
+    def _fix_typos_in_word(self, word, sym_spell):
+        """
+        Returns the corrected word if it has a correction, and the word not corrected otherwise
+        See details here:
+            https://symspellpy.readthedocs.io/en/latest/examples/lookup.html
+        """
+        correction = sym_spell.lookup(
+            word,
+            self.suggestion_verbosity,
+            self.distance,
+            transfer_casing=self.transfer_casing,
+            ignore_token=self.params["ignore_token"],
+        )
+        return correction[0].term if correction else word
+
+    def _fix_typos_in_document(self, string_or_token_list, sym_spell):
+        """
+        Returns the corrected document
+        See details here for word segmentation: 
+            https://symspellpy.readthedocs.io/en/latest/examples/word_segmentation.html
+        """
+        if not self.params["word_segmentation"]:
+            sent_fixed = []
+            for word in string_or_token_list:
+                if word not in self.params["set_untouched_words"] and not word.isdigit():
+                    sent_fixed.append(self._fix_typos_in_word(word, sym_spell))
+                else:
+                    sent_fixed.append(word)
+            return " ".join(sent_fixed)
+        else:
+            try:
+                # for some reason, word_segmentation got a stopIteration error. This is a symspellpy package issue.
+                return sym_spell.word_segmentation(
+                    string_or_token_list, ignore_token=self.params["ignore_token"]
+                ).corrected_string
+            except:
+                print("Couldn't segment the string:", string_or_token_list)
+                return string_or_token_list
+
+    def compute(self, df):
+
+        # new column names and matching dict
+        (
+            self.df_all_columns,
+            self.col_txt_lang_dict,
+            self.col_txt_preprocessed_dict,
+            self.col_txt_spellchecked_dict,
+        ) = create_all_new_column_names(df.columns, self.params["text_col_list"])
+
+        # Language detection
+        for col_txt, col_lang in self.col_txt_lang_dict.items():
+            df[col_lang] = get_language_from_column(df[col_txt], df.shape[0], self.params)
+
+        # Get all languages of the corpus
+        self.all_languages = self._get_all_languages(df)
+
+        # Preprocessint the text
+        preprocessing = PreprocessText(self.all_languages, self.params)
+        df = preprocessing.compute(df, self.col_txt_preprocessed_dict, self.col_txt_lang_dict)
+
+        # Create sym_spell objects
+        self.sym_spells = self._create_sym_spell_objects(df)
+
+        # Fix typos
+        print("Fixing typos ...")
+        for col_txt, col_spellchecked in self.col_txt_spellchecked_dict.items():
+            col_lang = self.col_txt_lang_dict[col_txt]
+            col_preprocessed = self.col_txt_preprocessed_dict[col_txt]
+            df[col_spellchecked] = df.apply(
```

Comment:
small enhancement suggestion: install tqdm and instead progress_apply instead of apply. This will give you a neat progress bar. Great for big datasets!